File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.java
class Solution {
    public String solution(int n, int t, int m, int p) {
        String answer = "", temp = "";
        
		// 게임 전체에서 나오는 값을 저장
        for(int i = 0; temp.length() < t*m; i++) {
            temp += Integer.toString(i, n);
        }
        
		// 원하는 순번에 외쳐야하는 번호만 따로 저장
        for(int i = p-1; answer.length() < t; i += m) {
            answer += temp.charAt(i);
        }
        
		// 모두 대문자로 반환
        return answer.toUpperCase();
    }
}

File: 프로그래머스/2/17687. ［3차］ n진수 게임/［3차］ n진수 게임.py
def convert_n(n, base):
    temp="0123456789ABCDEF"
    q,r = divmod(n, base)
    if q == 0:
        return temp[r]
    else:
        return convert_n(q, base) + temp[r]
def solution(n, t, m, p):
    answer = ''
    num = 0
    while True:  
        answer+=convert_n(num, n)
        if len(answer)>=t*m:
            answer = answer[:t*m]
            return answer[p-1:t*m:m]
        num+=1